packages/curve-ui-kit/src/features/user-profile/UserProfileButton.tsx

```diff
@@ -2,16 +2,17 @@ import PersonIcon from '@mui/icons-material/Person'
 import IconButton from '@mui/material/IconButton'
 import { AdvancedModeSwitch } from '@ui-kit/features/user-profile/settings/AdvancedModeSwitch'
 import { ThemeToggleButton } from '@ui-kit/features/user-profile/settings/ThemeToggleButton'
+import { useUserProfileButton } from '@ui-kit/hooks/useFeatureFlags'
 import { useSwitch } from '@ui-kit/hooks/useSwitch'
 import { t } from '@ui-kit/lib/i18n'
 import { UserProfile } from './UserProfile'
 
-export const UserProfileButton = ({ visible }: { visible: boolean }) => {
+export const UserProfileButton = () => {
   const [isOpen, open, close] = useSwitch(false)
   return (
     <>
-      {visible ? (
-        <IconButton size="small" onClick={open}>
+      {useUserProfileButton() ? (
```

Comment:
if you think we can get rid of `useUserProfileButton()` we could get rid of this ternary too